packages/nextjs/src/server/utils/getAuth.ts

```diff
@@ -0,0 +1,42 @@
+import { createSignedOutState } from '@clerk/backend-core';
+
+import type { RequestLike } from '../types';
+import { AuthData, AuthResult, SESSION_COOKIE_NAME, SessionsApi } from '../types';
+import { getAuthDataFromClaims } from './getAuthDataFromClaims';
+import { getAuthResultFromRequest, getCookie, getHeader } from './requestResponseUtils';
+
+export function createGetAuth(sessions: SessionsApi) {
+  return function getAuth(req: RequestLike): AuthData {
+    // When the auth result is set, we trust that the middleware has already run
+    // Then, we don't have to re-verify the JWT here,
+    // we can just strip out the claims manually.
+
+    const authResult = getAuthResultFromRequest(req);
+
+    if (!authResult) {
+      throw 'You need to use "withClerkMiddleware" in your Next.js middleware.js file. See https://nextjs.org/docs/advanced-features/middleware.';
```

Comment:
🔧 We should update this string to point to our Next.js docs.

Reply:
Done